### PR TITLE
Add DeploymentChain gRPCs

### PR DESCRIPTION
**Pull request by khanhtc1202**
**What this PR does / why we need it**:

This PR adds:
- ListDeploymentChains gRPC, used to fetch all deployment chains to show on list deployment chains page. There will be no options to filter deployment chain currently, just simply list all objects page by page.
- GetDeploymentChain gRPC, used to fetch a specified deployment chain by id to show on the deployment chain detailed page.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/app/api/grpcapi/web_api.go [hunk 1]
```diff
@@ -1781,3 +1783,79 @@ func (a *WebAPI) GetInsightApplicationCount(ctx context.Context, req *webservice
...
+	if errors.Is(err, datastore.ErrNotFound) {
+		return nil, status.Error(codes.NotFound, "Deployment chain is not found")
+	}
+	if err != nil {
+		a.logger.Error("failed to get deployment chain", zap.Error(err))
+		return nil, status.Error(codes.Internal, "Failed to get deployment chain")
+	}
+
+	if claims.Role.ProjectId != dc.ProjectId {
+		return nil, status.Error(codes.InvalidArgument, "Requested deployment chain does not belong to your project")
```
[the remark this answers is not in the document]
Comment (khanhtc1202) [reply pkg/app/api/grpcapi/web_api.go]: Sure, lets me change other places where we return `InvalidArgument` in the same case later by a separated PR.

[the remark this answers is not in the document]
Comment (khanhtc1202) [reply pkg/app/api/grpcapi/web_api.go]: For this PR, addressed by https://github.com/pipe-cd/pipe/pull/2923/commits/55a24239855d9d7dd64402517abdddeb1e34aeea

**Review by nakabonne — APPROVED** [review]
nice!

**Pull request merged**